src/java.base/share/classes/java/lang/constant/ConstantDescs.java

```diff
@@ -349,8 +350,11 @@ public static DirectMethodHandleDesc ofCallsiteBootstrap(ClassDesc owner,
                                                              String name,
                                                              ClassDesc returnType,
                                                              ClassDesc... paramTypes) {
-        return MethodHandleDesc.ofMethod(STATIC, owner, name, MethodTypeDesc.of(returnType, paramTypes)
-                                                                            .insertParameterTypes(0, INDY_BOOTSTRAP_ARGS));
+        int prefixLen = INDY_BOOTSTRAP_ARGS.length;
+        ClassDesc[] fullParamTypes = new ClassDesc[paramTypes.length + prefixLen];
+        System.arraycopy(INDY_BOOTSTRAP_ARGS, 0, fullParamTypes, 0, prefixLen);
+        System.arraycopy(paramTypes, 0, fullParamTypes, prefixLen, paramTypes.length);
```

Comment:
I'm thinking about creating a basic MethodTypeDesc like `INDY_BOOTSTRAP_TYPE = MethodTypeDesc.ofValidated(CD_Object, INDY_BOOTSTRAP_ARGS)`, and we derive bsm with 
```java
INDY_BOOTSTRAP_TYPE
    .insertParameterTypes(INDY_BOOTSTRAP_TYPE.parameterCount(), paramTypes)
    .changeReturnType(returnType)
```
which creates two MTD wrappers but they share the same parameter array

Reply:
That'd arguably be a bit cleaner. Might allocate an extra MTD (unless/until EA kicks in), but that might not matter really.